core/src/main/java/hudson/model/AbstractProject.java

```diff
@@ -1207,7 +1207,13 @@ public boolean checkout(AbstractBuild build, Launcher launcher, BuildListener li
             return true;    // no SCM
 
         FilePath workspace = build.getWorkspace();
-        workspace.mkdirs();
+        if(workspace!=null){
+            workspace.mkdirs();
+        }
+        else{
+            throw new AbortException(); // workspace is undefined
```

Comment:
The reason should be passed in the exception message. Otherwise the user will not see the root cause in the build log